libs/pika/async_mpi/include/pika/async_mpi/mpi_helpers.hpp

```diff
@@ -175,16 +133,49 @@ namespace pika::mpi::experimental::detail {
                 else
                 {
                     // pass the result onto a new task and invoke the continuation
-                    auto snd0 = ex::just(status) | ex::transfer(default_pool_scheduler(p)) |
-                        ex::then([receiver = PIKA_MOVE(receiver)](int status) mutable {
-                            PIKA_DETAIL_DP(
-                                mpi_tran<5>, debug(str<>("set_value_error_helper"), status));
-                            set_value_error_helper(status, PIKA_MOVE(receiver));
+                    auto snd0 = ex::transfer_just(default_pool_scheduler(p)) |
+                        ex::then([receiver = PIKA_MOVE(receiver)]() mutable {
+                            PIKA_DETAIL_DP(mpi_tran<5>, debug(str<>("set_value")));
+                            ex::set_value(PIKA_FORWARD(Receiver, receiver));
```

Comment:
Sorry, missed this earlier. The forward and move are in the wrong places:
```suggestion
                    auto snd0 = ex::transfer_just(default_pool_scheduler(p)) |
                        ex::then([receiver = PIKA_FORWARD(Receiver, receiver)]() mutable {
                            PIKA_DETAIL_DP(mpi_tran<5>, debug(str<>("set_value")));
                            ex::set_value(PIKA_MOVE(receiver));
```

Reply:
The receiver needs to be moved into the lambda, I'm not sure forward is right there, but it should be moved into set_value as well. 2 x `PIKA_MOVE` I think

Reply:
Your last change looks good to me. The first is a forward because `Receiver` is a template parameter of the with `Receiver&&` in the signature.

Reply:
I'm not sure I understand "Your last change looks good to me" contradicts "The first is a forward because Receiver is a template parameter of the with Receiver&& in the signature."
What are you trying to say? 
What I meant to say was yes - it is a template param that uses `&&`, but regardless of how it is passed into the callback registration function, it should be `moved` into the callback lambda, so I prefer move to forward - because then it really is moved. (probably the template should be changed to only accept a move....but it isn't worth the effort to be fair)

Reply:
What I meant is that your last change is what it should look like, followed by the reason why it should look like that, but you know that already. Hence I marked this as resolved.

Yes, in general we can start taking receivers by value, they have to be moved. But we can do that separately across all receivers, so let's leave it like this here.